src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -202,8 +201,10 @@ otError CslTxScheduler::HandleFrameRequest(Mac::TxFrame &aFrame)
 
     aFrame.SetChannel(mCslTxChild->GetCslChannel() == 0 ? Get<Mac::Mac>().GetPanChannel()
                                                         : mCslTxChild->GetCslChannel());
-    aFrame.SetTxPhase(mCslTxChild->GetCslPhase());
-    aFrame.SetTxPeriod(mCslTxChild->GetCslPeriod());
+    aFrame.SetTxDelay(mCslTxChild->GetCslTxDelay());
```

Comment:
Recalculate the delay with `GetNextCslTransmissionDelay`? I think recalculate the delay here will help,

1. Reduce memory usage as `mCslTxDelay` will not be needed?
2. Avoid the corner case when the code reaches here, the current time has just passed the saved `mCslTxDelay` and a new delay should be calculated.

Reply:
I think it's necessary to have both `TxDelay` and `TxDelayBaseTime` in a frame because when we finally call `nrf_802154_transmit_raw_at`, we need both parameters. Note that in this updated implementation, the base time for delay transmission is not `now`. I think there would be some problems if we do something like: add one to another and call the api with `0` and the sum.

Reply:
That's correct. Currently `nrf_802154_transmit_raw_at()` requires two parameters. I think it could work correctly if you implement a hack and pass:
`t0 = t0 + dt - 5000`
`dt = 5000`

But if this optimization is not critical for you now, I would recommend to use both parameters and wait until we optimize `nrf_802154_transmit_raw_at()` API. We plan to do that to reduce memory consumption inside the driver itself.

Reply:
What I meant is the member `mCslTxDelay` in class `ChildInfo` is not needed. It only exist because it avoids recalculation. But it brings the possibility of a corner case as stated above. So I would suggest to remove `mCslTxDelay` in `ChildInfo` and use `GetNextCslTransmissionDelay` to do recalculation here.